### PR TITLE
PLT-4257 Add pop-up asking if user wants to reset status

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Add pop-up asking if user wants to reset status if their status is manually set to away or offline.

@esethna, I didn't try it out on desktop app but I think it should work no problem

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4257

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

**Comment by jwilander** [comment]
Forgot to mention that we might want @asaadmahmood to touch up the look of the modal, such as the positioning of the checkbox, etc.

File: webapp/components/confirm_modal.jsx [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
+         */
+        confirmButtonText: PropTypes.node,
+
+        /*
+         * Text/jsx element on the cancel button
+         */
+        cancelButtonText: PropTypes.node,
+
+        /*
+         *
```
Comment (hmhealey) [inline webapp/components/confirm_modal.jsx]: You're missing a comment here

**Comment by esethna** [comment]
@jwilander please help address the conflicts and HH's comment when you have a chance?

**Comment by jwilander** [comment]
@hmhealey fixed

**Comment by jwilander** [comment]
By the way that test server should have worked despite the conflicts

**Comment by jwilander** [comment]
I don't think we need that with the coming change to show the status in the profile picture in the sidebar header. Also those ephemeral system messages are annoying

By the way I'd rather if we didn't use the awaiting submitter action label, it really clogs up the PR for no real reason

**Comment by esethna** [comment]
@asaadmahmood please help with the CSS changes when you have a chance today so we can get this in before code complete?

**Comment by esethna** [comment]
@asaadmahmood sorry there still seems to be some styling that is needed. There is an oddly large amount of space below the checkbox and above the line separator:

image

**Comment by esethna** [comment]
Looks good, thanks @asaadmahmood!

**Pull request merged**